src/tqec/block/block.py

```diff
@@ -266,49 +320,24 @@ def instantiate(self) -> cirq.Circuit:
         return circuit
 
     @override
-    def instantiate_without_boundary(self, boundary: BlockDimension) -> cirq.Circuit:
+    def instantiate_without_boundary(self, boundary: Direction3D) -> cirq.Circuit:
         return self.replace_boundary_with_empty_plaquettes(boundary).instantiate()
 
     @override
     def scale_to(self, k: int) -> None:
         self.template.scale_to(k)
 
-
-@dataclass
-class Computation:
-    """Represents a topological quantum error corrected computation.
-
-    The computation is represented by a mapping from a position to the
-    computational block whose origin is located at that position.
-    """
-
-    blocks: dict[Position3D, ComputationBlock]
-
-    def to_circuit(self) -> cirq.Circuit:
-        """Build and return the quantum circuit representing the computation.
-
-        Raises:
-            TQECException: if any of the circuits obtained by instantiating the
-                computational blocks is contains a qubit that is not a cirq.GridQubit.
-
-        Returns:
-            a cirq.Circuit instance representing the full computation.
-        """
-        instantiated_scheduled_blocks: list[ScheduledCircuit] = []
-        for position, block in self.blocks.items():
-
-            def shift_qubits(q: cirq.Qid) -> cirq.GridQubit:
-                if not isinstance(q, cirq.GridQubit):
-                    raise TQECException(
-                        f"Found a circuit with {q} that is not a cirq.GridQubit instance."
-                    )
-                return q + (position.x, position.y)
-
-            spatially_shifted_circuit = block.instantiate().transform_qubits(
-                shift_qubits
+    def _update_plaquettes(self, sides: list[TemplateSide]) -> None:
+        "Replaces plaquettes according to substitution rules."
+        for index in self.template.get_plaquette_indices_on_sides(sides):
+            replacement_index = self.template.get_corresponding_mid_for_side(index)
+            self.plaquettes.initial_plaquettes[index] = (
+                self.plaquettes.initial_plaquettes[replacement_index]
             )
-            instantiated_scheduled_blocks.append(
-                ScheduledCircuit(spatially_shifted_circuit, position.z)
-            )
-
-        return merge_scheduled_circuits(instantiated_scheduled_blocks)
+            self.plaquettes.final_plaquettes[index] = self.plaquettes.final_plaquettes[
+                replacement_index
+            ]
+            if self.repeating_plaquettes is not None:
```

Comment:
```suggestion
            if self.plaquettes.repeating_plaquettes is not None:
```
to silence typing error about `self.plaquettes.repeating_plaquettes` that may be `None`.